post_processing_stages/object_classify_tf_stage.cpp

```diff
@@ -0,0 +1,370 @@
+					ExecutionTime<std::micro>(std::bind(&ObjectClassifyTfStage::runInference, this)).count();
+				std::cout << time_taken << "ms" << std::endl;
+			}));
+		}
+	}
+
+	std::unique_lock<std::mutex> lock(output_results_mutex_);
+	std::string annotation = "Detected";
+
+	for (const auto &result : output_results_)
+	{
+		annotation = annotation + ", " + result.first + " : " + std::to_string(result.second);
+	}
+
+	completed_request.post_process_metadata.Set("object_classify.results", output_results_);
+
+	if (display_labels_ == 1)
+		completed_request.post_process_metadata.Set("annotate.text", annotation);
+
+	return false;
+}
+
+static PostProcessingStage *Create(LibcameraApp *app)
+{
+	return new ObjectClassifyTfStage(app);
+}
+
+static RegisterStage reg(NAME, &Create);
+
+void ObjectClassifyTfStage::runInference()
+{
+	int input_ = interpreter_->inputs()[0];
+	settings_.input_type = interpreter_->tensor(input_)->type;
+
+	for (int i = 0; i < tensor_input_.size(); i++)
+	{
+		interpreter_->typed_tensor<uint8_t>(input_)[i] = static_cast<uint8_t>(tensor_input_[i]);
+	}
+
+	if (interpreter_->Invoke() != kTfLiteOk)
+		throw std::runtime_error("Failed to invoke TFLite");
+
+	int output = interpreter_->outputs()[0];
+	TfLiteIntArray *output_dims = interpreter_->tensor(output)->dims;
+	// assume output dims to be something like (1, 1, ... ,size)
+	auto output_size = output_dims->data[output_dims->size - 1];
+
+	getTopResults(interpreter_->typed_output_tensor<uint8_t>(0), output_size, top_n_results_, threshold_high_,
+				  threshold_low_, &top_results_, settings_.input_type);
+
+	std::unique_lock<std::mutex> lock(output_results_mutex_);
+	output_results_.clear();
+
+	for (const auto &result : top_results_)
+	{
+		const float confidence = result.first;
+		const int index = result.second;
+		const std::string label = labels_[index];
+		output_results_.push_back(std::make_pair(label, confidence));
+	}
+}
+
+TfLiteStatus ObjectClassifyTfStage::readLabelsFile(const tflite::string &file_name, std::vector<tflite::string> *result,
+												   size_t *found_label_count)
+{
+	std::ifstream file(file_name);
+	if (!file)
+		throw std::runtime_error("Failed to load labels file");
+
+	result->clear();
+	tflite::string line;
+	while (std::getline(file, line))
+	{
+		result->push_back(line);
+	}
+
+	*found_label_count = result->size();
+	const int padding = 16;
+
+	while (result->size() % padding)
+	{
+		result->emplace_back();
+	}
+
+	return kTfLiteOk;
+}
+
+std::vector<uint8_t> ObjectClassifyTfStage::yuvToRgb(uint8_t *mem, int w, int h, int stride)
+{
+	uint8_t *Y = mem, *U = mem + h * stride, *V = mem + (5 / 4) * h * stride;
+	int dst;
+	std::vector<uint8_t> output(h * w * 3);
+
+	for (int y = 0; y < h; y++)
+	{
+		for (int x = 0; x < w; x++)
+		{
+			Y = mem + y * stride + x;
+			U = mem + (h * stride) + ((int)floor(y / 2)) * (stride / 2) + ((int)floor(x / 2));
+			V = mem + (5 * h * stride) / 4 + ((int)floor(y / 2)) * (stride / 2) + ((int)floor(x / 2));
```

Comment:
Is this just `V = U + h * stride / 4;` maybe?

Reply:
Yep, that works! changed it to the equation above.